core/platform/src/raw_syscalls.rs

```diff
@@ -3,242 +3,51 @@
-//     yield
-//     zero_arg_memop
-//     one_arg_memop
-//     four_arg_syscall
-//
-// If a new system call class that uses fewer than four arguments is added, then
-// the above list will need to be revised.
-//
-// Note that `command` always needs to use four_arg_syscall, even when calling a
-// command with fewer arguments, because we don't want to leak the
-// (possibly secret) values in the r2 and r3 registers to untrusted capsules.
-// Yield and memop do not have this concern and can leave arbitrary data in r2
-// and r3, because they are implemented by the core kernel, which is trusted.
-//
-// The success type for Memop calls depends on the operation performed. However,
-// all *currently defined* memop operations return either Success or Success
-// with u32. Therefore, the memop implementations only need to mark r0 and r1 as
-// clobbered, not r2 and r3. This choice of clobbers will need to be revisited
-// if and when a memop operation that returns more data is added.
-//
-// The decision of where to use u32 and usize can be a bit tricky. The Tock
-// syscall ABI is currently only specified for 32-bit systems, so on real Tock
-// systems both types match the size of a register, but the unit test
-// environment can be either 32 bit or 64 bit. This interface uses usize for
-// values that can contain pointers, so that pointers are not truncated in the
-// unit test environment. To keep types as consistent as possible, it uses u32
-// for all values that cannot be pointers.
 pub trait RawSyscalls {
-    // raw_yield should:
-    //     1. Call syscall class 0
-    //     2. Use register r0 for input and output as an inlateout register,
-    //        passing in r0_in and returning its value.
-    //     3. Mark all caller-saved registers as lateout clobbers.
-    //     4. NOT provide any of the following options:
-    //            pure             (yield has side effects)
-    //            nomem            (a callback can read + write globals)
-    //            readonly         (a callback can write globals)
-    //            preserves_flags  (a callback can change flags)
-    //            noreturn         (yield is expected to return)
-    //            nostack          (a callback needs the stack)
-    //
-    // Design note: This is safe because the yield types that currently exist
-    // are safe. If an unsafe yield type is added, we will need to make
-    // raw_yield unsafe. Although raw_yield shouldn't be called by code outside
-    // this crate, it can be, so that is a backwards-incompatible change. We
-    // pass YieldType rather than a usize because if we used usize directly then
-    // this API becomes unsound if the kernel adds support for an unsafe yield
-    // type (or even one that takes one more argument).
-    /// `raw_yield` should only be called by `libtock_platform`.
-    fn raw_yield(r0_in: YieldType) -> u32;
+    // Calls yield-no-wait, passing `flag` to the kernel. The kernel will set
```

Comment:
Should these be /// comments?

Reply:
I used `//` comments because this isn't really meant to be public-facing documentation. `RawSyscalls` is really internal to the `libtock-rs` library, I just don't have a way to express that using Rust's visibility specifiers. Except for the `clippy`-required `# Safety` comments, the only `///` comment tells not to use it directly.